### PR TITLE
Database setup

### LINKED ISSUE #32: Add request storage and sorting functionality
- [x] Database tables:
  - [x] Requests
  - [x] Trips
  - [x] RequestTrips   

- [x] Happy/sad path testing

- [x] Update to README (FE screenshots, BE API endpoint documented)

### PR DESCRIPTION
**Pull request by RetynaGirl**
Sets up database with lots of fancy tables and such, small change to API contract 

FRONT END WILL BREAK WITHOUT FE PR #84

Closes #32 

- [x] Sad Path Tested

- [x] Happy Path Tested

- [x] All Tests are passing

- [x] Updated README.md with added information

- [x] Collabotors:
- Brian Liu

### CODE REVIEW

File: app/poros/trip.rb [hunk 1]
```diff
@@ -1,55 +1,55 @@
...
-              :max_f,
-              :min_c,
-              :max_c,
-              :day_feels_like_c,
-              :day_feels_like_f,
-              :description,
-              :weather_date,
-              :weather,
-              :sky_coverage
+# class Trip
```
Comment (SageOfCode) [inline app/poros/trip.rb:1]: Why do we want our poros to be commented out?

File: app/models/trip.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+class Trip < ApplicationRecord
+  has_many :trip_weathers
+  has_many :weathers, through: :trip_weathers
+
+  has_many :request_trips
+  has_many :requests, through: :request_trips
+
+  def self.new_trip(flight, weather, index)
```
Comment (SageOfCode) [inline app/models/trip.rb:8]: Good work here.

File: app/serializers/trip_serializer.rb [hunk 3]
```diff
@@ -13,15 +13,47 @@ class TripSerializer
...
+  end
+
+  attributes :sky_coverage do |trip|
+    trip.weathers[0].sky_coverage
+  end
+
+  attributes :weather do |trip|
+    trip.weathers.map do |weather|
+      WeatherSerializer.new(weather)
+    end
```
Comment (JesseMellinger) [inline app/serializers/trip_serializer.rb:57]: Nice use of another serializer to format the data as expected within an array.

File: db/schema.rb [hunk 4]
```diff
@@ -10,9 +10,73 @@
...
+    t.datetime "date"
+    t.integer "sky_coverage"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
+  add_foreign_key "request_trips", "requests"
+  add_foreign_key "request_trips", "trips"
+  add_foreign_key "trip_weathers", "trips"
+  add_foreign_key "trip_weathers", "weathers"
```
Comment (JesseMellinger) [inline db/schema.rb:81]: Looks like two `join` tables then. Nice work.

File: app/models/trip.rb (hunk 2 above)
Comment (JesseMellinger) [reply app/models/trip.rb:8]: Looks like this effectively replaces the `initialize` method in our `Trip` poro. Nice!

File: app/poros/trip.rb (hunk 1 above)
Comment (JesseMellinger) [reply app/poros/trip.rb:1]: Because a `Trip` model has been created I think a PORO is unnecessary now.

**Review by JesseMellinger — APPROVED** [review]

**Review by Todd-Estes — APPROVED** [review]
Outstanding work, gentlemen !

File: app/models/trip.rb (hunk 2 above)
Comment (SageOfCode) [reply app/models/trip.rb:8]: Awesome, I didn't know if we wanted to delete these files if they are no longer being used.

**Pull request merged**